strutil/strutil.go

```diff
@@ -287,6 +298,18 @@ func RemoveEmpty(items []string) []string {
 	return itemsSlice
 }
 
+// RemoveGlobs removes any elements containing globs from a slice of strings.
+func RemoveGlobs(items []string) []string {
```

Comment:
my only doubt is this one on the other PR https://github.com/hashicorp/vault/pull/7277#discussion_r709503017 about if we need the `RemoveGlobs` function that we can discuss there, but other than that this PR looks good!

Reply:
Based on the discussion in the other PR, I've removed the RemoveGlobs func and squished the commit to make history nice and clean.